dm/syncer/dml.go

```diff
@@ -461,12 +473,13 @@ type DML struct {
 	originOldValues     []interface{} // only for update SQL
 	originValues        []interface{} // use to gen key and `WHERE`
 	safeMode            bool
-	key                 string // use to detect causality
-	downstreamIndexInfo *model.IndexInfo
+	key                 string                      // use to detect causality
+	downstreamIndexInfo *model.IndexInfo            // pick index from downstream which is pk/uk not null/uk value not null
```

Comment:
Add some comment like 'only use in genWhere'

Reply:
ok